platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,160 @@
+  private static final String FILE_PATH_2 = FILE_PATH + "2";
+
+  private static final String ENTITY_ID = "https://localhost:8993/services/saml/test";
+  private static final String ENTITY_ID_2 =
+      "http://localhost:80/spring-security-saml2-sample/saml/metadata";
+
+  private static final String REDIRECT_LOGOUT_URL =
+      "https://localhost:8993/services/saml/logout/redirect";
+  private static final String POST_LOGOUT_URL = "https://localhost:8993/services/saml/logout/post";
+  private static final String REDIRECT_ACS_URL =
+      "https://localhost:8993/services/saml/sso/redirect";
+  private static final String POST_ACS_URL = "https://localhost:8993/services/saml/sso/post";
+  private static final String SOAP_ACS_URL = "https://localhost:8993/services/saml/sso/soap";
+  private static final String CERTIFICATE =
+      "MIICsjCCAhugAwIBAgIGAWBQwru6MA0GCSqGSIb3DQEBCwUAMIGEMQswCQYDVQQGEwJVUzELMAkGA1UECBMCQVoxDDAKBgNVBAoTA0RERjEMMAoGA1UECxMDRGV2MRkwFwYDVQQDExBEREYgRGVtbyBSb290IENBMTEwLwYJKoZIhvcNAQkBFiJlbWFpbEFkZHJlc3M9ZGRmcm9vdGNhQGV4YW1wbGUub3JnMCAXDTE3MTIxMjE2NDM0N1oYDzIxMTcxMjEyMTY0MzQ3WjAUMRIwEAYDVQQDDAlsb2NhbGhvc3QwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDEwOqpKuMr+GmiT3kKcMKp4361ByVBaxvj5M6rptn+Al8oOZfxfaDOhOikUVuaPTuIYvTWW04KaQNHI3nm86RwldbwMlPZka5jGeYkOJ3qah1C6OMmMDEC7lM0/bhxPLc8C0cUwAG8FdlBLDKCYFjHj6UBq5xhD/1uJyvSoWcFqJQbXP+igVV7tDdRxRDhp8f5eActecjQXK5+bpzNs8KC+3gvnhBj/ux/DYyrE6iA4nKMnelIaZjhW11bsuQwM6XK8Tq7PQ2LuAuhFoKUq/p8p8Sva75/3nGeCHOlcjj7rnoSCcUgX+u1nnCN/wsshyRAX30wS7C75nQilQkxCfAlAgMBAAGjGDAWMBQGA1UdEQQNMAuCCWxvY2FsaG9zdDANBgkqhkiG9w0BAQsFAAOBgQAQBBked2MsqlmIwxSUXLee6f9R14GPsNsOkLohFhBTKuxC++n+q0+LvMRLmFxPj4dois/dHKQiD7wYWYY4WPvb3nD2aPev4NZm8erB9+XPVueRzxaLXQ+K1oW9Kjui8+2q1lTBBGOjma1DT/b6FVhyJo5SjQ3JMKe9TACVp/YA7A==";
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+
+    // read SPMetadata file into spMetadata
+    spMetadata = new ArrayList<>();
+    try (Stream<String> lines = Files.lines(Paths.get(FILE_PATH_2))) {
+      spMetadata.add(lines.collect(Collectors.joining(System.lineSeparator())));
+    }
+    try (Stream<String> lines = Files.lines(Paths.get(FILE_PATH))) {
+      spMetadata.add(lines.collect(Collectors.joining(System.lineSeparator())));
+    }
+
+    // set up binding set
+    bindingSet = ImmutableSet.of(Binding.HTTP_POST, Binding.HTTP_REDIRECT, Binding.SOAP);
+  }
+
+  @Test
+  public void parseNullMetadata() throws Exception {
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(null, bindingSet);
+
+    assertThat("Returns an empty map", testMap, is(equalTo(EMPTY_MAP)));
+  }
+
+  @Test
+  public void parseFirstMetadata() throws Exception {
+
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(spMetadata, bindingSet);
+
+    // get the first metadata
+    Iterator<Map.Entry<String, EntityInformation>> it = testMap.entrySet().iterator();
+    it.next();
+    Map.Entry<String, EntityInformation> testEntry = it.next();
```

Comment:
✏️ Should we just be getting the correct entity info based on the key of the map instead of iterating through the map? It just seems odd. And then you can verify that the `EntityInformation` retrieved is not null instead of asserting on the ID on line 96.

Reply:
Agree with @brjeter

Reply:
I did a refactor and I think this new way makes a lot of sense. I'll ping you guys when it's up.